File: MagicCollectionSpring/src/main/java/pga/magiccollectionspring/card/api/dto/CardSyncStatusDTO.java
package pga.magiccollectionspring.card.api.dto;

import java.time.LocalDateTime;
import java.util.List;

public record CardSyncStatusDTO(
        boolean scryfallInSync,
        boolean catalogStateMissing,
        LocalDateTime lastSyncedAt,
        String defaultCardsRemoteToken,
        String allCardsRemoteToken,
        List<LanguageSyncStatusDTO> languages
) {}

File: MagicCollectionAndroid/app/src/main/java/com/pga/magiccollection/data/remote/dto/CardSyncStatusDto.kt
package com.pga.magiccollection.data.remote.dto

data class CardSyncStatusDto(
    val scryfallInSync: Boolean,
    val catalogStateMissing: Boolean,
    val lastSyncedAt: String? = null,
    val defaultCardsRemoteToken: String? = null,
    val allCardsRemoteToken: String? = null,
    val languages: List<LanguageSyncStatusDto> = emptyList()
)

data class LanguageSyncStatusDto(
    val languageCode: String,
    val version: String? = null,
    val checksum: String? = null,
    val totalRows: Long = 0L,
    val status: String? = null
)
